src/GitObjectDb/Models/Merge/ObjectRepositoryMerge.cs

```diff
@@ -20,48 +19,43 @@ namespace GitObjectDb.Models.Merge
 {
     /// <inheritdoc/>
     [ExcludeFromGuardForNull]
-    public sealed class ObjectRepositoryMerge : IObjectRepositoryMerge
+    internal sealed class ObjectRepositoryMerge : IObjectRepositoryMerge
     {
-        private readonly IServiceProvider _serviceProvider;
         private readonly IRepositoryProvider _repositoryProvider;
         private readonly IModelDataAccessorProvider _modelDataProvider;
         private readonly MigrationScaffolderFactory _migrationScaffolderFactory;
+        private readonly MergeProcessor.Factory _mergeProcessorFactory;
+
         private readonly RepositoryDescription _repositoryDescription;
 
         /// <summary>
         /// Initializes a new instance of the <see cref="ObjectRepositoryMerge"/> class.
         /// </summary>
-        /// <param name="serviceProvider">The service provider.</param>
         /// <param name="container">The container.</param>
         /// <param name="repositoryDescription">The repository description.</param>
         /// <param name="repository">The repository on which to apply the merge.</param>
         /// <param name="mergeCommitId">The commit to be merged.</param>
         /// <param name="branchName">Name of the branch.</param>
-        /// <exception cref="ArgumentNullException">
-        /// serviceProvider
-        /// or
-        /// repositoryDescription
-        /// or
-        /// commitId
-        /// or
-        /// branchName
-        /// or
-        /// merger
-        /// </exception>
+        /// <param name="repositoryProvider">The repositoyr provider.</param>
+        /// <param name="modelDataProvider">The model data provider.</param>
+        /// <param name="migrationScaffolderFactory">The <see cref="MigrationScaffolder"/> factory.</param>
+        /// <param name="mergeProcessorFactory">The <see cref="MergeProcessor"/> factory.</param>
         [ActivatorUtilitiesConstructor]
-        public ObjectRepositoryMerge(IServiceProvider serviceProvider, IObjectRepositoryContainer container, RepositoryDescription repositoryDescription, IObjectRepository repository, ObjectId mergeCommitId, string branchName)
+        public ObjectRepositoryMerge(IObjectRepositoryContainer container, RepositoryDescription repositoryDescription, IObjectRepository repository, ObjectId mergeCommitId, string branchName,
```

Comment:
![Code Smell](https://sonarsource.github.io/sonar-github/code_smell.png 'Code Smell') Code Smell: Constructor has 9 parameters, which is greater than the 7 authorized. ([csharpsquid:S107](https://sonarcloud.io/organizations/frblondin-github/rules#rule_key=csharpsquid%3AS107))

[See it in SonarCloud](https://sonarcloud.io/project/issues?id=GitObjectDb&pullRequest=33&issues=AWe4KJcbD1JUB5M3-M2h&open=AWe4KJcbD1JUB5M3-M2h)

[ISSUE_KEY]: AWe4KJcbD1JUB5M3-M2h